backend/vxlan/vxlan.go

```diff
@@ -259,50 +250,68 @@ func (vb *VXLANBackend) handleSubnetEvents(batch []subnet.Event) {
 	}
 }
 
-func (vb *VXLANBackend) handleMiss(miss *netlink.Neigh) {
-	switch {
-	case len(miss.IP) == 0 && len(miss.HardwareAddr) == 0:
-		log.Info("Ignoring nil miss")
+func (vb *VXLANBackend) handleInitialSubnetEvents(batch []subnet.Event) {
+	log.Infof("Handling initial subnet events")
+	fdbTable, err := vb.dev.GetL2List()
+	l3Table, err1 := vb.dev.GetL3List()
 
-	case len(miss.IP) == 0:
-		vb.handleL2Miss(miss)
+	if err != nil {
```

Comment:
Do this check right after GetL2List() call. No need to do GetL3List() if it failed. You will also be able to re-use err variable.